### PR TITLE
Black the code

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
- Ran [black](https://github.com/python/black) on the codebase.
- Updated README
- **New** CONTRIBUTING.md file
- **New** LICENSE file

**Technical details:**
- CLI: (in usaspending-api dir) `$ black . `
    - Uses the config in pyproject.toml
    - ignores migration files
- Updated flake8 rules in setup.cfg
- Deleted .style.yapf
- Manual edits:
    - Since black will not alter code, strings there were on separate lines and combined onto a single line had an unnecessary `" "` between them. I removed these
    - A dictionary in city.py was manually edited to be more concise (Starting at line 108)
    - README.md

**Requirements for PR merge:**

1. [x] Unit & integration tests updated (N/A)
2. [x] API documentation updated (N/A)
3. [ ] Necessary PR reviewers:
    - [ ] Backend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed (N/A)
6. [x] Data validation completed (N/A)
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket (N/A)

**Area for explaining above N/A when needed:**
```
No API or database behavior or functionality should be changed by this PR
```

### CODE REVIEW

**Comment by BrianZito** [comment]
We will need to add documentation on using black, or else we can kiss dreams of this being an open source project goodbye...

**Comment by tony-sappe** [comment]
> We will need to add documentation on using black, or else we can kiss dreams of this being an open source project goodbye...

That sounds extremely dire. We certainly demand perfect code... except for that recent public PR with a typo in the comments.

**Comment by tony-sappe** [comment]
> On a serious note though, we should make a mention of Black on README.md (other than the checker tag) for the sake of contributors.

The current README doesn't contain any details for contributing. No mention of CodeClimate, flake8, expectations of the PR description, etc. If we wanted to store that info we should write a CONTRIBUTING.md file.

**Comment by BrianZito** [comment]
> > On a serious note though, we should make a mention of Black on README.md (other than the checker tag) for the sake of contributors.
> 
> The current README doesn't contain any details for contributing. No mention of CodeClimate, flake8, expectations of the PR description, etc. If we wanted to store that info we should write a CONTRIBUTING.md file.

PR descriptions are self-documented via the template, and CodeClimate and flake8 are checks in the pipeline, not tools expected to be run locally. Adding a CONTRIBUTING.md still might not be a bad idea though, if we feel we can maintain it and the client values that. I just don't want to see this bit slip through the cracks.

File: usaspending_api/accounts/models.py [hunk 1]
```diff
@@ -170,25 +167,20 @@ def totals(self):
             outlays[fiscal_year] += ab.gross_outlay_amount_by_tas_cpe
             obligations[fiscal_year] += ab.obligations_incurred_total_by_tas_cpe
         results = {
-            'outgoing': {
-                'outlays': outlays,
-                'obligations': obligations,
-                'budget_authority': budget_authority,
-            },
-            'incoming': {}
```
Comment (BrianZito) [inline usaspending_api/accounts/models.py]: Any chance we can disable JSON doing this? I haven't gotten that far into this PR yet, but this seems like a lot of unreadability coming our way...

Comment (tony-sappe) [reply usaspending_api/accounts/models.py]: This isn't JSON, it is a python dictionary. For this specific example it is less-readable. We can thank previous devs who decided to make it easy to read. Previously there weren't many requirements on how dictionaries were formatted, so there are plenty of other examples where black's format is more readable

Comment (BrianZito) [reply usaspending_api/accounts/models.py]: 20 files in I feel like the developers are winning most of these. I don't think going through each case will be productive; I just want to ask if you're sure that we can't just keep both formats and trust people to make smart decisions with dictionaries. I know black's whole idea is to be very strict, but this looks like a common thing we'll encounter.

**Comment by tony-sappe** [comment]
@BrianZito new CONTRIBUTING.md file was added to the PR

File: usaspending_api/accounts/models.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/accounts/models.py]: I suppose it is a personal preference. Yes, humans can make better decision for other humans over a program. Humans also make lots of mistakes, forget things, and have disagreements. Black is an opinionated code formatter. We'll remove the variability of the project's current code style in favor of a single format. If everyone hates it, we can re-evaluate.


```
    query = {
        "query": {
            "bool": {
                "must": [{
                    "match": {
                        "recipient_location_zip5": "abcde"
                    }
                }]
            }
        },
        "_source": ["award_id"]
    }```
vs
```
    query = {"query": {"bool": {"must": [{"match": {"recipient_location_zip5": "abcde"}}]}}, "_source": ["award_id"]}```

note: reducing the line count from 120 to ~90 will force more dictionaries to look like the "pretty print" style.

**Pull request merged**